src/Compiler/Checking/TailCallChecks.fsi

```diff
@@ -4,5 +4,6 @@ open FSharp.Compiler
 open FSharp.Compiler.TcGlobals
 open FSharp.Compiler.TypedTree
 
+/// Perform the TailCall analysis on the optimized TAST for a file.
```

Comment:
Nit: it might be relevant to summarize how this analysis is done.
Mentioning the attribute (because no `[<TailCall>]` no analysis) and what exactly happens with `implFileContents`. You traverse the tree, and if you found a function with the attribute you verify all its invocations?